rslib/src/import_export/text/csv/metadata.rs

```diff
@@ -176,36 +177,20 @@ impl Collection {
         }
     }
 
-    fn maybe_set_fallback_notetype(
+    pub(crate) fn maybe_set_notetype_and_deck(
         &mut self,
-        metadata: &mut CsvMetadata,
+        metadata: &mut crate::pb::import_export::CsvMetadata,
         notetype_id: Option<NotetypeId>,
+        deck_id: Option<DeckId>,
     ) -> Result<()> {
-        if let Some(ntid) = notetype_id {
-            metadata.notetype = Some(CsvNotetype::new_global(ntid));
-        } else if metadata.notetype.is_none() {
-            metadata.notetype = Some(CsvNotetype::new_global(self.fallback_notetype_id()?));
+        let defaults = self.defaults_for_adding(DeckId(0))?;
+        if metadata.notetype.is_none() {
```

Comment:
This should execute unconditionally if `notetype_id` is provided, or the user won't be able to change the notetype if one is specified in the imported file.